server/lib/base-logger.ts

```diff
@@ -0,0 +1,214 @@
+    });
+
+    // Prevents from exit on error related to the logger.
+    this.wazuhLogger.exitOnError = false;
+
+    // Plain text logger
+    this.wazuhPlainLogger = winston.createLogger({
+      level,
+      format: winston.format.simple(),
+      transports: [
+        new winston.transports.File({
+          filename: this.PLAIN_LOGS_PATH,
+        }),
+      ],
+    });
+
+    // Prevents from exit on error related to the logger.
+    this.wazuhPlainLogger.exitOnError = false;
+  };
+
+  /**
+   * Checks if wazuh/logs exists. If it doesn't exist, it will be created.
+   */
+  initDirectory = async () => {
+    try {
+      createDataDirectoryIfNotExists();
+      createDataDirectoryIfNotExists('logs');
+      if (typeof this.wazuhLogger === 'undefined' || typeof this.wazuhPlainLogger === 'undefined') {
+        this.initLogger();
+      }
+      this.allowed = true;
+      return;
+    } catch (error) {
+      this.allowed = false;
+      return Promise.reject(error);
+    }
+  };
+
+  /**
+   * Returns given file size in MB, if the file doesn't exist returns 0
+   * @param {*} filename Path to the file
+   */
+  getFilesizeInMegaBytes = (filename) => {
+    if (this.allowed) {
+      if (fs.existsSync(filename)) {
+        const stats = fs.statSync(filename);
+        const fileSizeInMegaBytes = stats.size;
+
+        return fileSizeInMegaBytes / 1000000.0;
+      }
+    }
+    return 0;
+  };
+
+  /**
+   * Check if file exist
+   * @param filename 
+   * @returns boolean 
+   */
+  checkFileExist = (filename) => {
+    return fs.existsSync(filename);
+  };
+
+  /**
+   * Checks if the wazuhapp.log file size is greater than 100MB, if so it rotates the file.
+   */
+  private checkFiles = () => {
+    if (this.allowed) {
+      // check raw log file
+      if (this.getFilesizeInMegaBytes(this.RAW_LOGS_PATH) >= 100) {
+        fs.renameSync(
+          this.RAW_LOGS_PATH,
+          `${WAZUH_DATA_LOGS_DIRECTORY_PATH}/wazuhapp.${new Date().getTime()}.log`
+        );
+        fs.writeFileSync(
+          this.RAW_LOGS_PATH,
+          JSON.stringify({
+            date: new Date(),
+            level: 'info',
+            location: 'logger',
+            message: 'Rotated log file',
+          }) + '\n'
+        );
+      }
+
+      // check log file
+      if (this.getFilesizeInMegaBytes(this.PLAIN_LOGS_PATH) >= 100) {
+        fs.renameSync(
+          this.PLAIN_LOGS_PATH,
+          `${WAZUH_DATA_LOGS_DIRECTORY_PATH}/wazuhapp-plain.${new Date().getTime()}.log`
+        );
+      }
+    }
+  };
+
+
+  /**
+   * Get Current Date
+   * @returns string
+   */
+  private yyyymmdd = () => {
+    const now = new Date();
+    const y = now.getFullYear();
+    const m = now.getMonth() + 1;
+    const d = now.getDate();
+    const seconds = now.getSeconds();
+    const minutes = now.getMinutes();
+    const hour = now.getHours();
+    return `${y}/${m < 10 ? '0' : ''}${m}/${d < 10 ? '0' : ''}${d} ${hour}:${minutes}:${seconds}`;
+  };
+
```

Comment:
Delete extra lines

Reply:
Actually this file doesn't have a prettier 🤓